src/zjs_callbacks.c

```diff
@@ -424,35 +414,39 @@ void zjs_call_callback(int32_t id, void* data, uint32_t sz)
 void zjs_service_callbacks(void)
 {
     if (ring_buf_initialized) {
-        uint16_t type;
-        uint8_t value;
-        uint8_t size = 0;
-        uint32_t dummy;
-        // setting size = 0 will check if there is an item in the ring buffer
-        while (zjs_port_ring_buf_get(&ring_buffer,
-                                     &type,
-                                     &value,
-                                     &dummy,
-                                     &size) == -EMSGSIZE) {
-            if (size == 0) {
-                break;
-            } else {
+        while (1) {
```

Comment:
Or maybe what we need to do is read all the callbacks out first and then call them, stopping when we get to the end. O_o